lib/Mojolicious/Plugin/ClientIP/Pluggable.pm

```diff
@@ -0,0 +1,179 @@
+package Mojolicious::Plugin::ClientIP::Pluggable;
+
+=head1 NAME
+
+Mojolicious::Plugin::ClientIP::CloudFlare - CloudFlare-aware client IP detection plugin
+
+=head1 SYNOPSIS
+
+    use Mojolicious::Lite;
+
+    plugin 'ClientIP::CloudFlare', v4_only => 1;
+    # for headers analisys customization
+    # plugin 'ClientIP::CloudFlare', analyzed_headers => [qw/cf-pseudo-ipv4 cf-connecting-ip/];
+
+
+    get '/' => sub {
+        my $c = shift;
+        $c->render(text => $c->client_ip);
+    };
+
+    app->start;
+
+=head1 DESCRIPTION
+
+Mojolicious::Plugin::ClientIP::CloudFlare is a Mojolicious plugin to get an IP address, possibly
+available beyound different CloudFlare-headers as well as X-Forwarded-For header.
+
+The pluging is inspired by L<Mojolicious::Plugin::ClientIP>.
+
+=head1 METHODS
+
+=head2 client_ip
+
+Find a client IP address from different CloudFlare headers, with fallback to C<X-Forwarded-For> header,
+and then to L<Mojo::Transaction#remote_address>. If the valid IP address is not found (if you are looking
+strictly for IPv4 address, but only IPv6-address is available via headers), it returns empty string.
+
+=head1 OPTIONS
+
+=head2 v4_only
+
+Only IPv4 addresses are considered valid among the possible addresses.
+
+    plugin 'ClientIP::CloudFlare', v4_only => 1;
+
+
+=head2 analyzed_headers
+
+Define order and names of CloudFlare-injected headers with client IP address.
+By default it uses C<cf-pseudo-ipv4 cf-connecting-ip true-client-ip>.
+
+    plugin 'ClientIP::CloudFlare', analyzed_headers => [qw/cf-pseudo-ipv4 cf-connecting-ip true-client-ip/].
+
+
+More details at L<https://support.cloudflare.com/hc/en-us/articles/202494830-Pseudo-IPv4-Supporting-IPv6-addresses-in-legacy-IPv4-applications>,
+L<https://support.cloudflare.com/hc/en-us/articles/200170986-How-does-CloudFlare-handle-HTTP-Request-headers>,
+L<https://support.cloudflare.com/hc/en-us/articles/206776727-What-is-True-Client-IP>
+
+=head1 COPYRIGHT AND LICENSE
+
+Copyright (C) 2017 binary.com
+
+=cut
+
+
+use strict;
+use warnings;
+
+use Data::Validate::IP;
+
+use Mojo::Base 'Mojolicious::Plugin';
+
+our $VERSION = '0.01';
+
+# for tests only
+use constant SKIP_LOOPBACK => $ENV{CLIENTIP_PLUGGABLE_SKIP_LOOPBACK} || 1;
+
+sub _check_ipv4 {
+    my ($ip) = @_;
+    return !Data::Validate::IP::is_unroutable_ipv4($ip)
```

Comment:
Why not `is_public_ipv4`?

Reply:
Actually I'm not sure now, that `is_public_*` is correct methods, as it fails to validate `203.0.113.43`.

Reply:
@basiliscos I think because of https://tools.ietf.org/html/rfc5735

Reply:
Indeed! Thanks! I fixed the code & tests.